js/main.js

```diff
@@ -58,19 +58,22 @@ function handleClick(event) {
     });
     xhr.send();
 
-    if (Object.entries(data.ratingsInfo).length === 0) {
+    if (data.ratingsInfo[collectionData.collectionName] === undefined) {
       data.ratingsInfo[collectionData.collectionName] = collectionData;
-    }
-
-    for (var key in data.ratingsInfo) {
-      if (key !== data.NFTvisible[0].collectionName) {
-        data.ratingsInfo.push(collectionData);
-      } else if (event.target.matches('.fa-heart')) {
-        data.ratingsInfo[key].likes++;
+      if (event.target.matches('.fa-heart')) {
+        data.ratingsInfo[collectionData.collectionName].likes++;
+      } else if (event.target.matches('.fa-star')) {
+        data.superliked.push(singleNFTData);
+      } else if (event.target.matches('.fa-thumbs-down')) {
+        data.ratingsInfo[collectionData.collectionName].dislikes++;
+      }
```

Comment:
You have this conditional in both conditional blocks, which means you do it either way. Instead of repeating yourself, you should just do this below the conditional on line 61